### PR TITLE
feat(@ngtools/webpack):  prevent build failure in case of custom locale

### PR DESCRIPTION
**Pull request by rahulbhooteshwar**
Fix for the issue #9772 . It will allow to use custom locale at build time. Locale data corresponding to custom locale can be registered using registerLocaleData in application module.

### CODE REVIEW

**Comment by rahulbhooteshwar** [comment]
I signed it!

**Comment by filipesilva** [comment]
@rahulbhooteshwar the error in CI is a flake. It's not caused by your PR and you can ignore it. Next time you push changes it should go away.

@ocombe can you weigh in on this topic?

**Comment by clydin** [comment]
Instead of the `console` usage, can you push the warning onto `this._warnings` so that the compilation system will direct the output to the configured location.

**Comment by ocombe** [comment]
Ah yes that's probably better!

**Comment by rahulbhooteshwar** [comment]
@ocombe I tried to incorporate changes suggested .  If we return `null ` in case of custom locale  causes another compiler error in my test application as following (My application i.e. testLocale is using custom-locale `no-no`):

`ERROR in ./src/main.ts
Module not found: Error: Can't resolve '@angular/common/locales/no-no' in 'F:\dev\testLocale\src'`

I think it has something to with the comment mentioned in the **angular_compiler_plugin.ts file** :

> it looks for the entry module  import, which will be replaced.

`// If we have a locale, auto import the locale data file.
      // This transform must go before replaceBootstrap because it looks for the entry module
      // import, which will be replaced.
      if (this._normalizedLocale) {
        this._transformers.push(registerLocaleData(isAppPath, getEntryModule,
          this._normalizedLocale));
      }`

**Comment by rahulbhooteshwar** [comment]
@ocombe  Thank you for taking time in reviewing and suggesting the best possible fix. 
I have incorporated changes suggested.

File: packages/@ngtools/webpack/src/angular_compiler_plugin.ts [hunk 1]
```diff
@@ -1056,9 +1056,12 @@ export class AngularCompilerPlugin {
         if (locales.indexOf(parentLocale) !== -1) {
           locale = parentLocale;
         } else {
-          throw new Error(
-            `Unable to load the locale data file "@angular/common/locales/${locale}", ` +
-            `please check that "${locale}" is a valid locale id.`);
+          this._warnings.push(`AngularCompilerPlugin: Unable to load the locale data file ` +
+            `"@angular/common/locales/${locale}", ` +
+            `please check that "${locale}" is a valid locale id.
+            Proceeding with default locale data registration i.e. 'en'.
```
Comment (ocombe) [inline packages/@ngtools/webpack/src/angular_compiler_plugin.ts]: you can remove `Proceeding with default locale data registration i.e. 'en'.` please

Comment (rahulbhooteshwar) [reply packages/@ngtools/webpack/src/angular_compiler_plugin.ts]: Done.

**Comment by rahulbhooteshwar** [comment]
@ocombe  I also need to modify test case in build-locale.ts  for the following line.

`.then(() => expectToFail(() => ng('build', '--aot', '--locale=no-locale')));`


I will be doing the change and will also amend the commit logs to comply with guidelines.

File: packages/@ngtools/webpack/src/angular_compiler_plugin.ts (hunk 1 above)
Comment (rahulbhooteshwar) [reply packages/@ngtools/webpack/src/angular_compiler_plugin.ts]: @ocombe  I am done  with all the  changes, including test case modification.

File: packages/@ngtools/webpack/src/angular_compiler_plugin.ts [hunk 2]
```diff
@@ -1056,9 +1056,11 @@ export class AngularCompilerPlugin {
         if (locales.indexOf(parentLocale) !== -1) {
           locale = parentLocale;
         } else {
-          throw new Error(
-            `Unable to load the locale data file "@angular/common/locales/${locale}", ` +
-            `please check that "${locale}" is a valid locale id.`);
+          this._warnings.push(`AngularCompilerPlugin: Unable to load the locale data file ` +
+            `"@angular/common/locales/${locale}", ` +
+            `please check that "${locale}" is a valid locale id.
+            If needed, localeData can be registered against custom locale in application module.`);
```
Comment (ocombe) [inline packages/@ngtools/webpack/src/angular_compiler_plugin.ts]: `If needed, localeData can be registered against custom locale in application module.` --> `If needed, you can use "registerLocaleData" manually.`

**Comment by ocombe** [comment]
Just a small change to the text of the warning and it should be good

File: packages/@ngtools/webpack/src/angular_compiler_plugin.ts (hunk 2 above)
Comment (rahulbhooteshwar) [reply packages/@ngtools/webpack/src/angular_compiler_plugin.ts]: Sure will do  it asap.

**Comment by rahulbhooteshwar** [comment]
@filipesilva  can you please check the **continuous-integration/travis-ci/pr** job failure. I am unable to get it!!

**Comment by rahulbhooteshwar** [comment]
@ocombe apologies ! I amended last commit log  without thinking that it will dismiss your review/approval. Please approve and take it forward  again.

File: packages/@ngtools/webpack/src/angular_compiler_plugin.ts (hunk 2 above)
Comment (ocombe) [reply packages/@ngtools/webpack/src/angular_compiler_plugin.ts]: you still need to change this text :)

**Pull request merged**